### PR TITLE
Fix #230: Make REPORT_PERCENTAGE a float

### LINKED ISSUE #230: [Beta] Report percentage cannot be smaller than 1%
Previously `django-csp<4` had float values 0–1 that allowed for fractions or setting the reporting percentage below 1% generally. Now it is 0–100 in `4.0b1`:

> `REPORT_PERCENTAGE`
> - Percentage of requests that should see the report-uri directive. Use this to throttle the number of CSP violation reports made to your report-uri. An **_integer_** between 0 and 100 (0 = no reports at all). Ignored if report-uri isn’t set.

The migration assistant does `round(n*100)` so if originally the sample was 0.25% this would round it to 0 effectively turning off reporting. _(Sure, it's for the consumers to figure this out! The migration is just a help, it still should be reviewed manually to catch similar issues…)_

Is there a technical limitation to keep the new %% value only as integer from now on? A high traffic site may make use of reporting settings smaller than 1% [[e.g.](https://github.com/mozilla/bedrock/issues/14453#issuecomment-2194643867)] so allowing floats would actually solve <1% values generally, and also allow keeping the same sample size migrating from previous versions.

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -110,10 +110,9 @@ policy.
        on the same origin.
 
 ``REPORT_PERCENTAGE``
-    Percentage of requests that should see the ``report-uri`` directive.
-    Use this to throttle the number of CSP violation reports made to your
-    ``report-uri``. An **integer** between 0 and 100 (0 = no reports at all).
-    Ignored if ``report-uri`` isn't set.
+    Percentage of requests that should see the ``report-uri`` directive.  Use this to throttle the
+    number of CSP violation reports made to your ``report-uri``. A **float** between 0.0 and 100.0
```
Comment (stevejalim) [inline docs/configuration.rst:114]: > A **float** between 0.0 and 100.0

Won't a float of `100.0` give us a REPORT_PERCENTAGE of 10000.0? Would be good to warn about invalid values maybe?

```
        config["REPORT_PERCENTAGE"] = _REPORT_PERCENTAGE * 100
```

File: csp/contrib/rate_limiting.py [hunk 2]
```diff
@@ -23,7 +23,8 @@ def get_policy_parts(self, request: HttpRequest, response: HttpResponseBase, rep
         if policy is None:
             return policy_parts
 
-        remove_report = random.randint(0, 99) >= policy.get("REPORT_PERCENTAGE", 100)
+        # `random.random` returns a value in the range [0.0, 1.0) so all values will be < 100.0.
```
Comment (stevejalim) [inline csp/contrib/rate_limiting.py:26]:
```suggestion
        # `random.random` returns a value in the range [0.0, 1.0] so all values will be < 100.0.
```

File: docs/configuration.rst (hunk 1 above)
Comment (stevejalim) [reply docs/configuration.rst:114]: I think it'd be nice, to help avoid surprises, but can be a follow-on thing or a ticket for someone in the community to pick up?

**Review by stevejalim — APPROVED** [review]
r+

File: csp/contrib/rate_limiting.py (hunk 2 above)
Comment (robhudson) [reply csp/contrib/rate_limiting.py:26]: That was actually intentional to denote that 1.0 isn't included in the range that `random.random()` can return.

**Pull request merged**